packages/adapters/wagmi/src/client.ts

```diff
@@ -138,28 +145,26 @@ export class WagmiAdapter implements ChainAdapter {
 
   public constructor(
     configParams: Partial<CreateConfigParameters> & {
-      networks: CaipNetwork[]
+      networks: AppKitNetwork[]
       projectId: string
     }
   ) {
     if (!configParams.projectId) {
       throw new Error(ErrorUtil.ALERT_ERRORS.PROJECT_ID_NOT_CONFIGURED.shortMessage)
     }
 
-    this.caipNetworks = configParams.networks.map(caipNetwork => ({
-      ...caipNetwork,
-      rpcUrl: CaipNetworksUtil.extendRpcUrlWithProjectId(caipNetwork.rpcUrl, configParams.projectId)
-    }))
+    this.caipNetworks = CaipNetworksUtil.extendCaipNetworks(configParams.networks, {
+      projectId: configParams.projectId,
+      customNetworkImageUrls: {}
+    }) as [CaipNetwork, ...CaipNetwork[]]
 
-    this.wagmiChains = convertToAppKitChains(
-      this.caipNetworks.filter(
-        caipNetwork => caipNetwork.chainNamespace === CommonConstantsUtil.CHAIN.EVM
-      )
-    )
+    this.wagmiChains = this.caipNetworks.filter(
+      caipNetwork => caipNetwork.chainNamespace === CommonConstantsUtil.CHAIN.EVM
+    ) as unknown as [BaseNetwork, ...BaseNetwork[]]
 
     const transportsArr = this.wagmiChains.map(chain => [
       chain.id,
-      getTransport({ chain, projectId: configParams.projectId })
+      getTransport({ chain: chain as Chain, projectId: configParams.projectId })
```

Comment:
What is `Chain`? We might need to use types from common/TypeUtils

Reply:
`Chain` is viem's chain type. 
If this is the same type for other networks we can keep, else we should replace for our own type